lib/streamlit/testing/element_tree.py

```diff
@@ -711,6 +713,124 @@ def set_range(self, lower: T, upper: T) -> SelectSlider[T]:
+    type: str
+    id: str
+    label: str
+    max_chars: int
+    help: str
+    form_id: str
+    autocomplete: str
+    placeholder: str
+    disabled: bool
+    key: str | None
+
+    root: ElementTree = field(repr=False)
+
+    def __init__(self, proto: TextInputProto, root: ElementTree):
+        self.proto = proto
+        self.root = root
+        self._value = None
+
+        self.type = "text_input"
+        self.id = proto.id
+        self.label = proto.label
+        self.max_chars = proto.max_chars
+        self.help = proto.help
+        self.form_id = proto.form_id
+        self.autocomplete = proto.autocomplete
+        self.placeholder = proto.placeholder
+        self.disabled = proto.disabled
+        self.key = user_key_from_widget_id(self.id)
+
+    def set_value(self, v: str) -> TextInput:
+        self._value = v
+        return self
+
+    def widget_state(self) -> WidgetState:
+        ws = WidgetState()
+        ws.id = self.id
+        ws.string_value = self.value
+        return ws
+
+    @property
+    def value(self) -> str:
+        if self._value is not None:
+            return self._value
+        else:
+            state = self.root.session_state
+            assert state
+            # Awkward to do this with `cast`
+            return state[self.id]  # type: ignore
+
+    def input(self, v: str) -> TextInput:
+        # TODO should input be setting or appending?
+        if self.max_chars and len(v) > self.max_chars:
+            return self
+        return self.set_value(v)
+
+
+@dataclass(repr=False)
+class TextArea(Element):
+    _value: str | None
+    proto: TextAreaProto
+    type: str
+    id: str
+    label: str
+    max_chars: int
+    help: str
+    form_id: str
+    placeholder: str
+    disabled: bool
+    key: str | None
+
+    root: ElementTree = field(repr=False)
+
+    def __init__(self, proto: TextAreaProto, root: ElementTree):
+        self.proto = proto
+        self.root = root
+        self._value = None
+
+        self.type = "text_area"
+        self.id = proto.id
+        self.label = proto.label
+        self.max_chars = proto.max_chars
+        self.help = proto.help
+        self.form_id = proto.form_id
+        self.placeholder = proto.placeholder
+        self.disabled = proto.disabled
+        self.key = user_key_from_widget_id(self.id)
+
+    def set_value(self, v: str) -> TextArea:
+        self._value = v
+        return self
+
+    def widget_state(self) -> WidgetState:
+        ws = WidgetState()
+        ws.id = self.id
+        ws.string_value = self.value
+        return ws
+
+    @property
+    def value(self) -> str:
+        if self._value is not None:
+            return self._value
+        else:
+            state = self.root.session_state
+            assert state
+            # Awkward to do this with `cast`
+            return state[self.id]  # type: ignore
+
+    def input(self, v: str) -> TextArea:
+        # TODO should input be setting or appending?
```

Comment:
Good question, I personally think the setting is fine, to be similar to `check` for checkbox and `select` for selection. 

But on the other hand `select` add an option on multi-select :) 
I think it should be definitely appending if we call it something like "type", but for input assignment sounds good for me.